torchbenchmark/models/nvidia_deeprecommender/nvinfer.py

```diff
@@ -0,0 +1,254 @@
+  parser.add_argument('--non_linearity_type', type=str, default="selu", metavar='N',
+                      help='type of the non-linearity used in activations')
+  parser.add_argument('--save_path', type=str, default="autorec.pt", metavar='N',
+                      help='where to save model')
+  parser.add_argument('--predictions_path', type=str, default="out.txt", metavar='N',
+                      help='where to save predictions')
+  parser.add_argument('--batch_size', type=int, default=1, metavar='N',
+                      help='inference batch size')
+  parser.add_argument('--jit', action='store_true',
+                      help='jit-ify model before running')
+  parser.add_argument('--forcecuda', action='store_true',
+                      help='force cuda use')
+  parser.add_argument('--forcecpu', action='store_true',
+                      help='force cpu use')
+  parser.add_argument('--nooutput', action='store_true',
+                      help='disable writing output to file')
+  parser.add_argument('--silent', action='store_true',
+                      help='disable output messages')
+  parser.add_argument('--profile', action='store_true',
+                      help='enable profiler and stat print')
+  
+  args = parser.parse_args()
+
+  return args
+
+def getBenchmarkArgs(forceCuda):
+
+  class Args:
+    pass
+  
+  args = Args()
+
+  args.drop_prob          = 0.8
+  args.constrained        = False
+  args.skip_last_layer_nl = False
+  args.hidden_layers      = '512,512,1024'
+  args.path_to_train_data = os.path.dirname(__file__) + '/Netflix/N1W_TRAIN'
+  args.path_to_eval_data  = os.path.dirname(__file__) + '/Netflix/N1W_TEST'
+  args.non_linearity_type = 'selu'
+  args.save_path          = 'model_save/model.epoch_0'
+  args.predictions_path   = 'preds.txt'
+  args.batch_size         = 1
+  args.jit                = False
+  args.forcecuda          = forceCuda
+  args.forcecpu           = not forceCuda
+  args.nooutput           = True
+  args.silent             = True
+  args.profile            = False
+
+  return args
+
+def processArgState(args) :
+
+  if not args.silent:
+    print(args)
+  
+  if args.forcecpu and args.forcecuda:
+      print("Error, force cpu and cuda cannot both be set")
+      quit()
+  
+  args.use_cuda = torch.cuda.is_available() # global flag
+  if not args.silent:
+    if args.use_cuda:
+      print('GPU is available.') 
+    else: 
+      print('GPU is not available.')
+  
+  if args.use_cuda and args.forcecpu:
+      args.use_cuda = False
+  
+  if not args.silent:
+    if args.use_cuda:
+      print('Running On GPU')
+    else:
+      print('Running On CUDA')
+  
+    if args.profile:
+      print('Profiler Enabled')
+
+  return args
+
+class DeepRecommenderInferenceBenchmark:
+
+  def __init__(self, device = 'cpu', jit=False, usecommandlineargs = False) :
+
+    self.toytest = True
+
+    if self.toytest:
+      self.toyinputs = torch.randn(1,15178).to(device)
```

Comment:
If we are using randomly generated toyinputs, why we are still checking in the input files such as n1w.test.txt?